src/runtime/hexagon_remote/sim_remote.cpp

```diff
@@ -311,6 +311,18 @@ int initialize_kernels(const unsigned char *code, int codeLen, bool use_dlopenbu
             halide_print(NULL, dlerror());
             return -1;
         }
+
+        if (trace_lib_base) {
+            void *fun_ptr = dlsym(lib, "plt_halide_malloc");
+            Dl_info dli;
+            int rc = dladdr((void *)fun_ptr, &dli);
+            if (rc == 0) {
+                printf("rc = %d\n", rc);
```

Comment:
Seems like this should be one fprintf statement, with %d as the error code, e.g.:

    fprintf(stderr, "dladdr failed (%d)\n", rc);